### PR TITLE
Provide improvements for our database schema

### PR DESCRIPTION
**Pull request by lachmanfrantisek**
preview: https://github.com/lachmanfrantisek/research/blob/database-schema-update/database/README.md

### CODE REVIEW

**Review by TomasTomecek — COMMENTED** [review]
Wow, this is interesting, the step model seems to be a fitting abstraction to our needs. On the other hand, the final pipelines will be really complex but we can't escape that I'd say.

File: database/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+Currently, user can request a rerun for a single chroot using the `retry` button. With that, there will be a new pipeline for each chroot we re-trigger. To be able to group all the retried chroots, we can create a [so-called requested action](https://docs.github.com/en/rest/guides/getting-started-with-the-checks-api#check-runs-and-requested-actions) to re-trigger all or all failed. We can also provide choise of the step to re-trigger.
+
+## Effectiveness of the queries
+
+For the regular queries covering multiple entries that goes across the database, it might be better to create them as one query and not relly on the ORM.
+
+### Migrations
+
+- For the model that groups the Copr chroots together, we need to group them by `copr_id` or `SRPMBuildModel`.
+- In case of `TF`, `SRPMBuildModel` is probably the only way, but this doesn't cover the test retrigger. We can use `TFTTestRunModel.submitted_time`.
```
Comment (lbarcziova) [inline database/README.md:72]: could you please include that for tests without the build phase the `SRPMBuildModel` cannot be used as well?

File: database/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+Currently, user can request a rerun for a single chroot using the `retry` button. With that, there will be a new pipeline for each chroot we re-trigger. To be able to group all the retried chroots, we can create a [so-called requested action](https://docs.github.com/en/rest/guides/getting-started-with-the-checks-api#check-runs-and-requested-actions) to re-trigger all or all failed. We can also provide choise of the step to re-trigger.
+
+## Effectiveness of the queries
+
+For the regular queries covering multiple entries that goes across the database, it might be better to create them as one query and not relly on the ORM.
+
+### Migrations
+
+- For the model that groups the Copr chroots together, we need to group them by `copr_id` or `SRPMBuildModel`.
```
Comment (lbarcziova) [inline database/README.md:71]: this should be the same for the Koji build, right?

**Review by lbarcziova — COMMENTED** [review]
nice work!

**Review by mfocko — COMMENTED** [review]
lgtm; sorry for late review, I had it open and didn't finish the last few sections

**Pull request merged**